internal/formats/common/cyclonedxhelpers/format.go

```diff
@@ -0,0 +1,90 @@
+package cyclonedxhelpers
+
+import (
+	"time"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/version"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	"github.com/google/uuid"
+)
+
+func ToFormatModel(s sbom.SBOM) *cyclonedx.BOM {
+	cdxBOM := cyclonedx.NewBOM()
+	versionInfo := version.FromBuild()
+
+	cdxBOM.SerialNumber = uuid.New().URN()
+	cdxBOM.Metadata = toBomDescriptor(internal.ApplicationName, versionInfo.Version, s.Source)
+
+	packages := s.Artifacts.PackageCatalog.Sorted()
+	components := make([]cyclonedx.Component, len(packages))
+	for i, p := range packages {
+		components[i] = toComponent(p)
+	}
+	cdxBOM.Components = &components
```

Comment:
I'm not sure if it's equivlant, but if line 23 does
```
cdxBOM.Components = make([]cyclonedx.Component, len(packages))
```

We can cut out the middle man and just append directly to the struct?
```
cdxBOM.Components = appened(cdxBOM.Components, toComponent(p))
```

This is where I don't have a preference really, but am curious about what people think from a general programming sense. Especially if append vs index assignment is faster when using Go.

Reply:
In general I'm a +0 in the category of performance between these options... for readability I'm a +1 for your suggestion, @spiffcs in general. However, in this case I don't think that'll work; the field on the struct isn't a slice, it's a pointer to a slice:
```
type BOM struct {
	...
	Components         *[]Component         `json:"components,omitempty" 
        ...
}
```